resources/ec2_security_groups.go

```diff
@@ -406,23 +390,43 @@ func fetchEc2SecurityGroupIpPermissions(ctx context.Context, meta schema.ClientM
 	if !ok {
 		return fmt.Errorf("not ec2 security group")
 	}
-	res <- securityGroup.IpPermissions
+	type ipPermission struct {
+		types.IpPermission
+		permissionType string
+	}
+
+	ipRanges := make([]ipPermission, len(securityGroup.IpPermissionsEgress)+len(securityGroup.IpPermissions))
+	for i, ip := range securityGroup.IpPermissionsEgress {
+		ipRanges[i] = ipPermission{ip, "egress"}
+	}
+	lenEgressRules := len(securityGroup.IpPermissionsEgress)
+	for i, ip := range securityGroup.IpPermissions {
+		ipRanges[i+lenEgressRules] = ipPermission{ip, "ingress"}
+	}
+	res <- ipRanges
 	return nil
 }
 func fetchEc2SecurityGroupIpPermissionIpRanges(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
 	securityGroupIpPermission, ok := parent.Item.(types.IpPermission)
 	if !ok {
 		return fmt.Errorf("not ec2 security group ip permission")
 	}
-	res <- securityGroupIpPermission.IpRanges
-	return nil
-}
-func fetchEc2SecurityGroupIpPermissionIpv6Ranges(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
-	securityGroupIpPermission, ok := parent.Item.(types.IpPermission)
-	if !ok {
-		return fmt.Errorf("not ec2 security group ip permission")
+
+	type customIpRange struct {
+		cidr        string
+		description string
+		cidr_type   string
+	}
+
+	ipRanges := make([]customIpRange, len(securityGroupIpPermission.IpRanges)+len(securityGroupIpPermission.Ipv6Ranges))
```

Comment:
maybe just use `make([]T, 0, capacity)` and use `append()` below to simplify the reading and avoid any future errors.

Reply:
Sorry didn't mean to include this change in this pr. I have removed it, but will apply your changes to my local branch